File: frontend/src/types/project.ts
// Types pour les projets
export interface Project {
  id: string
  name: string
  main_website?: string
  description?: string
  created_at: string
  updated_at: string
  keywords: ProjectKeyword[]
  competitors: Competitor[]
}

export interface ProjectCreate {
  name: string
  main_website?: string
  description?: string
  keywords?: string[]
}

export interface ProjectUpdate {
  name?: string
  main_website?: string
  description?: string
}

export interface ProjectSummary {
  id: string
  name: string
  main_website?: string
  description?: string
  created_at: string
  updated_at: string
  keywords_count: number
  competitors_count: number
  analyses_count: number
}

export interface Competitor {
  id: string
  project_id: string
  name: string
  website: string
  created_at: string
}

export interface CompetitorCreate {
  name: string
  website: string
}

export interface ProjectKeyword {
  project_id: string
  keyword: string
  created_at: string
} 

File: backend/app/schemas/project.py
from typing import List, Optional
from pydantic import HttpUrl, Field
from .base import BaseCreateSchema, BaseUpdateSchema, BaseReadSchema

# Schémas pour les mots-clés
class KeywordCreate(BaseCreateSchema):
    keyword: str = Field(..., min_length=1, max_length=100)

class KeywordRead(BaseCreateSchema):
    keyword: str

# Schémas pour les concurrents
class CompetitorCreate(BaseCreateSchema):
    name: str = Field(..., min_length=1, max_length=200)
    website: str = Field(..., min_length=1, max_length=500)

class CompetitorUpdate(BaseUpdateSchema):
    name: Optional[str] = Field(None, min_length=1, max_length=200)
    website: Optional[str] = Field(None, min_length=1, max_length=500)

class CompetitorRead(BaseReadSchema):
    project_id: str
    name: str
    website: str

# Schémas pour les projets
class ProjectCreate(BaseCreateSchema):
    name: str = Field(..., min_length=1, max_length=200, description="Nom du projet")
    main_website: Optional[str] = Field(None, max_length=500, description="Site web principal")
    description: Optional[str] = Field(None, max_length=1000, description="Description du projet")
    keywords: List[str] = Field(default=[], description="Liste des mots-clés cibles")

class ProjectUpdate(BaseUpdateSchema):
    name: Optional[str] = Field(None, min_length=1, max_length=200)
    main_website: Optional[str] = Field(None, max_length=500)
    description: Optional[str] = Field(None, max_length=1000)
    keywords: Optional[List[str]] = Field(None, description="Liste des mots-clés cibles")

class ProjectRead(BaseReadSchema):
    name: str
    main_website: Optional[str]
    description: Optional[str]
    keywords: List[KeywordRead] = []
    competitors: List[CompetitorRead] = []

class ProjectSummary(BaseReadSchema):
    """Version simplifiée pour les listes"""
    name: str
    main_website: Optional[str]
    description: Optional[str]
    keywords_count: int = 0
    competitors_count: int = 0
    analyses_count: int = 0 